### ISSUE TITLE
CentOS 6 compatibility

### ISSUE THREAD

**Issue by dimitri** [edicl/cl-fad#9]
Current version (as in Quicklisp latest release) of cl-fad is not usable in CentOS 6 with SBCL out of the box, because it's using the `sb-ext:delete-directory` function that appeared in SBCL 1.0.44.

I wonder how easy it is to add compatibility to that. If that's simple enough, then I think it's worth the trouble.

A user of my software (pgloader, which depends on cl-fad) reported some details:

```
[rjgonzale@cloud1 pgloader]$ cat /etc/redhat-release 
CentOS release 6.4 (Final)

from epel I installed

[rjgonzale@cloud1 pgloader]$ sbcl --version
SBCL 1.0.38-3.el6

And the error is...

[rjgonzale@cloud1 pgloader]$ ./pgloader.lisp --help
; Upgrading ASDF from version 2.010 to version 2.26
Loading quicklisp and the pgloader project and its dependencies...
;
; compilation aborted because of fatal error:
;   SB-INT:SIMPLE-READER-PACKAGE-ERROR at 13483 (line 282, column 30)
on
#<SB-SYS:FD-STREAM ;
for
"file /home/rjgonzale/quicklisp/dists/quicklisp/software/cl-fad-0.7.2/fad.lisp" ;
{10042479D1}>: ;     Symbol "DELETE-DIRECTORY" not found in the SB-EXT
package.
```

**Comment by amitlan** [comment]
Was this ever addressed maybe elsewhere?

**Comment by hanshuebner** [comment]
Not to my knowledge.  I'd recommend using a version of SBCL that is less
ancient.

2015-01-21 11:25 GMT+01:00 Amit Langote <notifications@github.com>:

**Comment by dimitri** [comment]
Yeah I managed to have a build machine with a recent enough SBCL, and to then only deploy the binary artefact given by buildapp, so that I don't have such a pressing need for a backward compatible solution in cl-fad.

Also, I have to admit that I now use uiop pathnames facilities rather than anything else, because of the way asdf and uiop are distributed and mainly already available (in modern enough versions of SBCL, again).

**Comment by Hellseher** [comment]
I've got the same issue when try to compile CL-FAD on CentOS 6

    ; Loading "cl-fad"
.   ;
   ; compilation aborted because of fatal error:
   ;   SB-INT:SIMPLE-READER-PACKAGE-ERROR at 13483 (line 282, column 30) on #<SB-SYS:FD-STREAM
   ;                                                                          for "file /root/quicklisp/dists/quicklisp/software/cl-fad-0.7.3/fad.lisp"
   ;                                                                          {1003A91E91}>:
   ;     Symbol "DELETE-DIRECTORY" not found in the SB-EXT package.   


    #: sbcl --version
    SBCL 1.0.38-3.el6

And struggling to compile new version on SBCL...

**Comment by stassats** [comment]
That SBCL version is even older now, even less incentive to support it.

**Comment by Hellseher** [comment]
Any clue how to compile last one using old SBCL 1.0.38? So far I've got error.

**Comment by dimitri** [comment]
Just upgrade your SBCL to something less ancient, really.
